.github/workflows/integration-tests.yml

```diff
@@ -311,6 +315,91 @@ jobs:
+
+      - name: Generate port
+        id: generate-port
+        env:
+          GITHUB_PR_NUMBER: ${{ github.event.number }}
+        run: |
+          PORT_BASE=3001
+          MAX_PORT=8000 
+  
+          # Use PR number as offset. Given GitHub PRs are incremental, this guarantees uniqueness for at least 5000 PRs.
+          OFFSET=$GITHUB_PR_NUMBER
+          echo "PR Number: $OFFSET"
+  
+          # Ensure that we don't exceed the max port
+          if (( OFFSET > (MAX_PORT - PORT_BASE) )); then
+              OFFSET=$((OFFSET % (MAX_PORT - PORT_BASE)))
+          fi
+  
+          # Map the offset to the port range
+          REMOTE_PORT=$((PORT_BASE + OFFSET))
+          echo "REMOTE_PORT=$REMOTE_PORT" >> $GITHUB_OUTPUT
+      - name: Reverse SSH Tunneling
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.image.tag-suffix == '-plugins'
+        env:
+          TRACING_SSH_KEY: ${{ secrets.TRACING_SSH_KEY }}
+          TRACING_SSH_SERVER: ${{ secrets.TRACING_SSH_SERVER }}
+          REMOTE_PORT: ${{ steps.generate-port.outputs.REMOTE_PORT }}
+        run: |
+            eval $(ssh-agent)
+            echo "test"
+            echo "$TRACING_SSH_KEY" | wc -c
+            echo "$TRACING_SSH_KEY" | tr -d '\r' | wc -c
+            echo "$TRACING_SSH_KEY" | tr -d '\r' | base64 --decode | ssh-add -
+            # f: background process
+            # N: do not execute a remote command
+            # R: remote port forwarding
+            ssh -o StrictHostKeyChecking=no -f -N -R $REMOTE_PORT:localhost:3000 user-gha@$TRACING_SSH_SERVER
+            echo "To view Grafana locally:"
+            echo "ssh -N -L 8000:localhost:$REMOTE_PORT user-gha@$TRACING_SSH_SERVER"
+            echo "Then visit http://localhost:8000 in a browser."
+            echo "If you are unable to connect, check with the security team that you have access to the tracing server."
+      - name: Show Grafana Logs
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.image.tag-suffix == '-plugins'
+        run: |
+            docker logs grafana
+      - name: Show Tempo Logs
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.image.tag-suffix == '-plugins'
+        run: |
+            docker logs tempo
+      - name: Show OpenTelemetry Collector Logs
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.image.tag-suffix == '-plugins'
+        run: |
+            docker logs otel-collector
+      - name: Set sleep time to use in future steps
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.image.tag-suffix == '-plugins'
+        run: |
+          echo "SLEEP_TIME=2400" >> "$GITHUB_ENV"
```

Comment:
what is 2400 and why?

Reply:
Amount of time the step is kept running, which is the amount of time the traces are stored in the github runner and available for access. 

I think that should be clear enough in how it's used in future steps, but if it's not, we can rename.